Jenkinsfile

```
@@ -30,15 +30,35 @@ pipeline {
     quietPeriod(10)
   }
   triggers {
-    issueCommentTrigger("(${obltGitHubComments()}|^run (compatibility|benchmark|integration) tests)")
+    issueCommentTrigger("(${obltGitHubComments()}|^run (jdk compatibility|benchmark|integration|end-to-end) tests)")
   }
   parameters {
     string(name: 'MAVEN_CONFIG', defaultValue: '-V -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Dhttps.protocols=TLSv1.2 -Dmaven.wagon.http.retryHandler.count=3 -Dmaven.wagon.httpconnectionManager.ttlSeconds=25', description: 'Additional maven options.')
+
+    // Note about GH checks and optional steps
+    //
+    // All the steps executed by default are included in GH checks
+    // All the mandatory steps not included by default need to be added in GH branch protection rules.
+
+    // enabled by default, required for merge through default GH check
     booleanParam(name: 'test_ci', defaultValue: true, description: 'Enable Unit tests')
-    booleanParam(name: 'agent_integration_tests_ci', defaultValue: true, description: 'Enable Agent Integration tests')
-    booleanParam(name: 'endtoend_tests_ci', defaultValue: true, description: 'Enable APM End-to-End Integration tests')
-    booleanParam(name: 'bench_ci', defaultValue: true, description: 'Enable benchmarks')
-    booleanParam(name: 'compatibility_ci', defaultValue: false, description: 'Enable JDK compatibility tests')
+
+    // disabled by default, but required for merge, there are two GH checks:
+    // - Non-Application Server integration tests
+    // - Application Server integration tests
+    // opt-in with 'ci:agent-integration'
+    booleanParam(name: 'agent_integration_tests_ci', defaultValue: false, description: 'Enable Agent Integration tests')
+
+    // disabled by default, but required for merge, GH check name is ${GITHUB_CHECK_ITS_NAME}
+    // opt-in with 'ci:end-to-end' tag on PR
+    booleanParam(name: 'endtoend_tests_ci', defaultValue: false, description: 'Enable APM End-to-End tests')
```

To be aligned with the label and stage name:

```suggestion
    booleanParam(name: 'end_to_end_tests_ci', defaultValue: false, description: 'Enable APM End-to-End tests')
```